File: frontend/src/types/pharmacy.ts
export type TherapeuticCategory =
  | "ANALGESIC"
  | "ANTIBIOTIC"
  | "ANTIMALARIAL"
  | "ANTIVIRAL"
  | "IV_FLUIDS"
  | "RESPIRATORY"
  | "CARDIOVASCULAR"
  | "VACCINE"
  | "EMERGENCY_DRUG"

export type ExpiryStatus = "HEALTHY" | "APPROACHING" | "CRITICAL" | "EXPIRED"

export interface MedicineBatch {
  batch_id: string
  batch_number: string
  medicine_id: string
  facility_id: string
  quantity: number
  manufacturing_date: string
  expiry_date: string
  days_to_expiry: number
  expiry_status: ExpiryStatus
  unit_cost_inr: number
  is_cold_chain: boolean
  supplier_name: string
}

export interface MedicineSKU {
  id: string
  name: string
  generic_name: string
  category: TherapeuticCategory
  dosage_form: string
  strength: string
  unit: string
  min_buffer_stock: number
  reorder_point: number
  total_stock: number
  daily_burn_rate: number
  days_of_supply: number
  is_essential: boolean
  is_cold_chain: boolean
  batches_count: number
  stock_status: "SAFE" | "REORDER_NEEDED" | "CRITICAL_STOCKOUT"
}

export interface BatchAllocation {
  batch_id: string
  batch_number: string
  expiry_date: string
  quantity_dispensed: number
  remaining_in_batch: number
}

export interface DispenseResult {
  transaction_id: string
  medicine_id: string
  medicine_name: string
  total_quantity_dispensed: number
  allocations: BatchAllocation[]
  facility_id: string
  dispensed_at: string
  dispensed_by: string
  fefo_compliant: boolean
}

export interface ExpiryRiskReport {
  critical_batches_count: number
  approaching_batches_count: number
  estimated_financial_risk_inr: number
  at_risk_batches: MedicineBatch[]
}

File: backend/app/schemas/pharmacy.py
from enum import Enum
from typing import Optional, List
from pydantic import BaseModel, Field
from datetime import date, datetime

class TherapeuticCategory(str, Enum):
    ANALGESIC = "ANALGESIC"
    ANTIBIOTIC = "ANTIBIOTIC"
    ANTIMALARIAL = "ANTIMALARIAL"
    ANTIVIRAL = "ANTIVIRAL"
    IV_FLUIDS = "IV_FLUIDS"
    RESPIRATORY = "RESPIRATORY"
    CARDIOVASCULAR = "CARDIOVASCULAR"
    VACCINE = "VACCINE"
    EMERGENCY_DRUG = "EMERGENCY_DRUG"

class ExpiryStatus(str, Enum):
    HEALTHY = "HEALTHY"          # > 90 days
    APPROACHING = "APPROACHING"  # 31 - 90 days
    CRITICAL = "CRITICAL"        # <= 30 days
    EXPIRED = "EXPIRED"          # <= 0 days

class MedicineBatch(BaseModel):
    batch_id: str
    batch_number: str
    medicine_id: str
    facility_id: str
    quantity: int
    manufacturing_date: str
    expiry_date: str
    days_to_expiry: int
    expiry_status: ExpiryStatus
    unit_cost_inr: float
    is_cold_chain: bool = False
    supplier_name: str

class MedicineSKU(BaseModel):
    id: str
    name: str
    generic_name: str
    category: TherapeuticCategory
    dosage_form: str  # Tablet, Syrup, Injection, Vial, Sachet
    strength: str
    unit: str         # strips, bottles, vials, sachets
    min_buffer_stock: int
    reorder_point: int
    total_stock: int
    daily_burn_rate: float
    days_of_supply: float
    is_essential: bool = True
    is_cold_chain: bool = False
    batches_count: int
    stock_status: str  # "SAFE", "REORDER_NEEDED", "CRITICAL_STOCKOUT"

class DispenseRequest(BaseModel):
    facility_id: str
    medicine_id: str
    quantity: int
    patient_id: Optional[str] = None
    doctor_name: Optional[str] = None
    notes: Optional[str] = None

class BatchAllocation(BaseModel):
    batch_id: str
    batch_number: str
    expiry_date: str
    quantity_dispensed: int
    remaining_in_batch: int

class DispenseResult(BaseModel):
    transaction_id: str
    medicine_id: str
    medicine_name: str
    total_quantity_dispensed: int
    allocations: List[BatchAllocation]
    facility_id: str
    dispensed_at: str
    dispensed_by: str
    fefo_compliant: bool = True

class ExpiryRiskReport(BaseModel):
    critical_batches_count: int
    approaching_batches_count: int
    estimated_financial_risk_inr: float
    at_risk_batches: List[MedicineBatch]
